### DISCUSSION TITLE
Scheduled Habits

### DISCUSSION THREAD

**Discussion by yohannd1** [iSoron/uhabits#454] [Feature Requests]
Hello, I would like to suggest **Scheduled Habits** on Loop Habit Tracker. Here's how they could work:
1. The user can schedule days where the habit is obligatory.
2. These habits will have the checkboxes highlighted on the scheduled days.
2.1. If the day passes and the user didn't check off, the habits can have an "X" on the checkbox.

Also, some extra features related to this:
1. Option to filter only habits scheduled for that day.
2. Home Screen Widget with the habits for that day (and only one checkbox for each habit).

**Comment by zuron7** [comment]
+1 Habit notifications if stickied should automatically disappear at the end of the day or should have an option to mark not done.

**Comment by eSkry** [comment]
+1 Good idea!

**Comment by ke352802081770314** [comment]
I disagree with 2.1 - it should not be mandatory to maintain habit tracker in real time, so recording achievements after the fact should not be prevented. (This also helps new users to back-fill the history from other records they have, rather than having to start from scratch)

**Reply by yohannd1** [reply]
I'm not completely sure what I meant when I wrote that (it's been a while), but as far as I remember the "X" is just a way to indicate the habit hasn't been completed and it should have been. This wouldn't prevent recording them after they are due.

Now that I think of it, though, it'd be good if the indicator had a starting date (by default its date of creation) so days before it's been created don't get marked as incomplete. Not sure if this is that necessary, though.

**Reply by ke352802081770314** [reply]
Maybe I misinterpreted your intention. I only started with Loop Habit Tracker version 2.0 and I understand from older tickets that the UI was substantially different previously.
For habit forming, a starting date has no influence. only performing a habit counts, whether I previously didn't even plan to do something has no benefit on the scoring. Currently, if "use question mark for unfilled dates" (my translation back from my UI language, may be inaccurate) is set, nevertheless by my understanding scoring works the same for ? and × marks.
